File: DiKErnel/Test/DiKErnel.FunctionLibrary.Test/GrassRevetmentWaveRunup/GrassRevetmentWaveRunupRepresentative2PInputTest.cpp
// Copyright (C) Stichting Deltares 2022. All rights reserved.
//
// This file is part of DiKErnel.
//
// DiKErnel is free software: you can redistribute it and/or modify
// it under the terms of the GNU Lesser General Public License as published by
// the Free Software Foundation, version 3.
// 
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
// GNU Lesser General Public License for more details.
//
// You should have received a copy of the GNU Lesser General Public License
// along with this program. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.
//
// All names, logos, and references to "Deltares" are registered trademarks of
// Stichting Deltares and remain full property of Stichting Deltares at all times.
// All rights reserved.

#include <gtest/gtest.h>

#include "GrassRevetmentWaveRunupRepresentative2PInput.h"

namespace DiKErnel::FunctionLibrary::Test
{
    TEST(GrassRevetmentWaveRunupRepresentative2PInputTest, GivenInput_WhenCreated_ThenExpectedValues)
    {
        // Given & When
        constexpr GrassRevetmentWaveRunupRepresentative2PInput input{};

        // Then
        ASSERT_DOUBLE_EQ(0.0, input._surfSimilarityParameter);
        ASSERT_DOUBLE_EQ(0.0, input._waveAngleImpact);
        ASSERT_DOUBLE_EQ(0.0, input._waveHeightHm0);
        ASSERT_DOUBLE_EQ(0.0, input._representativeWaveRunup2PGammab);
        ASSERT_DOUBLE_EQ(0.0, input._representativeWaveRunup2PGammaf);
        ASSERT_DOUBLE_EQ(0.0, input._representativeWaveRunup2PAru);
        ASSERT_DOUBLE_EQ(0.0, input._representativeWaveRunup2PBru);
        ASSERT_DOUBLE_EQ(0.0, input._representativeWaveRunup2PCru);
    }
}

File: Test/DiKErnel.FunctionLibrary.Test/GrassRevetmentWaveRunup/GrassRevetmentWaveRunupRepresentative2PInputTest.cs
// Copyright (C) Stichting Deltares and State of the Netherlands 2023. All rights reserved.
//
// This file is part of DiKErnel.
//
// This program is free software; you can redistribute it and/or modify it under the terms of
// the GNU Lesser General Public License as published by the Free Software Foundation; either
// version 3 of the License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
// without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
// See the GNU Lesser General Public License for more details.
//
// You should have received a copy of the GNU Lesser General Public License along with this
// program; if not, see <https://www.gnu.org/licenses/>.
//
// All names, logos, and references to "Deltares" are registered trademarks of Stichting
// Deltares and remain full property of Stichting Deltares at all times. All rights reserved.

using DiKErnel.FunctionLibrary.GrassRevetmentWaveRunup;
using DiKErnel.TestUtil;
using NUnit.Framework;

namespace DiKErnel.FunctionLibrary.Test.GrassRevetmentWaveRunup
{
    [TestFixture]
    public class GrassRevetmentWaveRunupRepresentative2PInputTest
    {
        [Test]
        public void Constructor_ExpectedValues()
        {
            // Setup
            double waveAngle = Random.NextDouble();
            double waveAngleImpact = Random.NextDouble();
            double waveHeightHm0 = Random.NextDouble();
            double representativeWaveRunup2PGammab = Random.NextDouble();
            double representativeWaveRunup2PGammaf = Random.NextDouble();
            double representativeWaveRunup2PAru = Random.NextDouble();
            double representativeWaveRunup2PBru = Random.NextDouble();
            double representativeWaveRunup2PCru = Random.NextDouble();

            // Call
            var input = new GrassRevetmentWaveRunupRepresentative2PInput(waveAngle, waveAngleImpact,
                                                                         waveHeightHm0, representativeWaveRunup2PGammab,
                                                                         representativeWaveRunup2PGammaf,
                                                                         representativeWaveRunup2PAru,
                                                                         representativeWaveRunup2PBru,
                                                                         representativeWaveRunup2PCru);

            // Assert
            Assert.That(input.WaveAngle, Is.EqualTo(waveAngle));
            Assert.That(input.WaveAngleImpact, Is.EqualTo(waveAngleImpact));
            Assert.That(input.WaveHeightHm0, Is.EqualTo(waveHeightHm0));
            Assert.That(input.RepresentativeWaveRunup2PGammab, Is.EqualTo(representativeWaveRunup2PGammab));
            Assert.That(input.RepresentativeWaveRunup2PGammaf, Is.EqualTo(representativeWaveRunup2PGammaf));
            Assert.That(input.RepresentativeWaveRunup2PAru, Is.EqualTo(representativeWaveRunup2PAru));
            Assert.That(input.RepresentativeWaveRunup2PBru, Is.EqualTo(representativeWaveRunup2PBru));
            Assert.That(input.RepresentativeWaveRunup2PCru, Is.EqualTo(representativeWaveRunup2PCru));
        }
    }
}